File: paper/src/main/java/net/lib/luxorigin/LanguageAPI/LanguageAPIHandler.java
package net.lib.luxorigin.LanguageAPI;

import com.google.gson.Gson;
import com.google.gson.GsonBuilder;
import lombok.Getter;
import net.lib.luxorigin.LanguageAPI.utils.FormatingHelper;
import org.bukkit.plugin.java.JavaPlugin;

import java.io.File;
import java.io.InputStream;
import java.util.Enumeration;
import java.util.HashMap;
import java.util.Locale;
import java.util.Map;
import java.util.jar.JarEntry;
import java.util.jar.JarFile;

public final class LanguageAPIHandler {

    // NOOP
    private LanguageAPIHandler() {}

    private static final Gson gson = new GsonBuilder().setPrettyPrinting().create();

    @Getter
    private static JavaPlugin registered;
    @Getter
    private static final Map<JavaPlugin, Map<String, Map<String, String>>> cache = new HashMap<>();

    @Getter
    private static String language = Locale.getDefault().getLanguage().toLowerCase();

    public static void register(JavaPlugin plugin) {
        try {
            Map<String, Map<String, String>> langs = new HashMap<>();
            File jar = new File(plugin.getClass().getProtectionDomain().getCodeSource().getLocation().toURI());

            try (JarFile jarFile = new JarFile(jar)) {
                Enumeration<JarEntry> entries = jarFile.entries();

                while (entries.hasMoreElements()) {
                    JarEntry entry = entries.nextElement();
                    String path = entry.getName();

                    if (!path.startsWith("lang/") && !path.startsWith("languages/") && !path.startsWith("language/") || entry.isDirectory()) {
                        continue;
                    }
                    String file = path.substring(path.indexOf("/") + 1);
                    String locale = file.substring(0, file.lastIndexOf("."));
                    String ext = file.substring(file.lastIndexOf(".") + 1);

                    try (InputStream stream = plugin.getResource(path)) {
                        if (stream == null) continue;

                        if (langs.containsKey(locale)) {
                            throw new RuntimeException("Duplicated locale : " + locale);
                        }

                        langs.put(locale, switch (ext) {
                            case "json" -> FormatingHelper.parseJson(gson, stream);
                            case "yml", "yaml" -> FormatingHelper.parseYaml(stream);
                            case "ini", "lang" -> FormatingHelper.parseINI(stream);
                            default -> Map.of();
                        });
                    } catch (Exception e) {
                        throw new RuntimeException("Language load failed", e);
                    }
                }
            }
            cache.put(plugin, langs);
            registered = plugin;
        } catch (Exception e) {
            throw new RuntimeException("Language load failed", e);
        }
    }

    public static void setLanguage(String language) {
        language = language.toLowerCase();

        if (cache.getOrDefault(registered, Map.of()).containsKey(language)) {
            LanguageAPIHandler.language = language;
        }
    }
}

File: pmmp/src/lib/luxorigin/LanguageAPI/LanguageAPIHandler.php
<?php

declare(strict_types=1);

namespace lib\luxorigin\LanguageAPI;

use lib\luxorigin\LanguageAPI\utils\FormatingHelper;
use pocketmine\plugin\PluginBase;

final class LanguageAPIHandler
{
    /** @var PluginBase|null */
    protected static ?PluginBase $registered = null;

    /** @var array<string, array<string, array<string, string>>> */
    protected static array $languages = [];

    /** @var string */
    protected static string $language = "en";

    // NOOP
    private function __construct() {}

    public static function register(PluginBase $plugin): void
    {
        self::$registered = $plugin;
        self::$language = $plugin->getServer()->getLanguage()->getLang();

        foreach (["lang", "languages", "language"] as $folder) {
            $directory = rtrim($plugin->getDataFolder(), "/\\") . DIRECTORY_SEPARATOR . $folder;

            if (!is_dir($directory)) continue;

            foreach (glob($directory . DIRECTORY_SEPARATOR . "*") ?: [] as $filename) {
                self::$languages = FormatingHelper::load(self::$languages, $plugin->getName(), $filename);
            }
        }

        foreach ($plugin->getResources() as $path => $resource) {
            if (!preg_match("#^(lang|language|languages)/#", $path)) continue;

            $ext = pathinfo($path, PATHINFO_EXTENSION);
            $extension = ["json", "yml", "yaml", "ini", "lang"];

            if (!in_array($ext, $extension, true)) {
                continue;
            }
            $target = rtrim($plugin->getDataFolder(), "/\\") . DIRECTORY_SEPARATOR . $path;
            $locale = strtolower(pathinfo($target, PATHINFO_FILENAME));

            if (isset(self::$languages[$plugin->getName()][$locale])) continue;

            if (!file_exists($target)) {
                $plugin->saveResource($target);
            }
            self::$languages = FormatingHelper::load(self::$languages, $plugin->getName(), $target);
        }
    }

    public static function getRegistered(): ?PluginBase
    {
        return self::$registered ?? throw new \LogicException("Plugin registered not found");
    }

    public static function isRegistered(): bool
    {
        return self::$registered instanceof PluginBase;
    }

    public static function getLanguage(): string
    {
        return self::$language ?? "en";
    }

    public static function getCache(): array
    {
        return self::$languages;
    }

    public static function get(string $key): string
    {
        return self::$languages[self::getRegistered()->getName()][self::getLanguage()][$key] ?? $key;
    }

    public static function setLanguage(string $language): void
    {
        $language = strtolower($language);

        if (!isset(self::$languages[self::getRegistered()->getName()][$language])) {
            return;
        }
        self::$language = $language;
    }
}